core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -342,7 +340,7 @@ private BufferChainOutputStream() {
 
     @Override
     public void write(int b) throws IOException {
-      singleByte[0] = (byte) b;
+      byte[] singleByte = new byte[]{(byte)b};
```

Comment:
How about lazy initialize instead? Unless we think Java would optimize out this allocation.